runtime/sema/check_casting_expression.go

```diff
@@ -412,3 +421,204 @@ func FailableCastCanSucceed(subType, superType Type) bool {
 
 	return true
 }
+
+// IsCastRedundant checks whether a simple cast is redundant.
+// Checks for two cases:
+//    - Case I: Contextually expected type is same as casted type (target type).
+//    - Case II: Expression is self typed, and is same as the casted type (target type).
+func IsCastRedundant(expr ast.Expression, exprInferredType, targetType, expectedType Type) bool {
+	if expectedType != nil && expectedType.Equal(targetType) {
```

Comment:
While Cadence doesn't currently have type aliases, we can probably future proof this against such a feature by changing this `equal` check to instead check whether both `expectedType` and `targetType` are subtypes of each other. At the moment this seems like it will only give the redundant cast hint when the two types are syntactically equal, but not when they are semantically equal.

Reply:
Hmm, that's a good point! But that might depend on how we define the `equal` method for type aliases. Like for e.g: In Go, type aliases are 'equal' at the user-level, and it would come down to how we are going to implement it.
We use this equal method across the checker, and all such places might affect when we introduce such a feature.